numpy/ma/core.py

```diff
@@ -219,7 +219,7 @@ def default_fill_value(obj):
         defval = default_filler['i']
     elif isinstance(obj, str):
         defval = default_filler['S']
-    elif isinstance(obj, unicode):
+    elif sys.version_info[0] < 3 and isinstance(obj, unicode):
```

Comment:
Yes. If you look in `numpy/compat/py3k.py` you will see that both unicode and bytes are defined so as to cover the correct types in both python 2 and python 3.